### ISSUE TITLE
Unify with novus/nvd3 repo

### ISSUE THREAD

**Issue by joshjordan** [liquidpele/nvd3#45]
Yeah, I know what you're thinking: "novus dropped this and doesn't want to maintain it". Yes, Novus did drop it and left some people stranded on a library that wasn't moving forward. And that is Bad. And you forked it, and kept it moving. And that is Awesome.

Unfortunately, there are still folks who are confused about which repo to use, or that simply don't know about `nvd3-community`. Worse, now the two repos are diverging, and we'll end up with some features & bugs in one, and other features & bugs in the other.

I think we can work together to get the best of both worlds. I can get a commitment from Novus `nvd3` a priority if we can get our ducks in a row. @liquidpele, are you open to having a conversation about it?

**Comment by liquidpele** [comment]
Sure :)

How do you imagine things being set up?

**Comment by joshjordan** [comment]
Want to hit me up by email at josh.jordan@gmail.com?

**Comment by robinfhu** [comment]
Hey can you guys keep me in the loop as well. Thanks.

**Comment by robinfhu** [comment]
Alright, I did a quick test of this build, and I think it'll work with our software.  I like some of the stuff in here.  I think we should merge this into novus/nvd3 master.

**Comment by robinfhu** [comment]
Merged branch created: https://github.com/novus/nvd3/tree/massive-merge

There are some things I need to do to make it ready:
* Fix the examples and test pages (some are broken).
* Make sure unit tests work.
* Make sure this version of nvd3 works with our internal software. 

But otherwise, @liquidpele seriously amazing work.

**Comment by portante** [comment]
@robinfhu, @liquidpele, can you two lay out a vision for what the state of nvd3 will be after this merge?  Will folks be working mainly from the community version?  Or from the Novus github repo instead?

**Comment by liquidpele** [comment]
So basically it sounds like novus hired some people and want to work on the code again.   Robin is apparently merging in all the code from nvd3-community into branches within their novus repo right now.  They are also discussing adding me as a contributor to the main novus repo, but we're just discussing via email right now.  

Personally I don't care what repo is used, but I don't want to be limited by novus politics so we'll see :)

**Comment by robinfhu** [comment]
@portante @liquidpele , I'd like for the two forks to remain in sync in the short term.  I'd like to think a bit more about whether we can keep the community version and just merge things from that every once in a while.

**Comment by portante** [comment]
@liquidpele, @robinfhu, I think what might help here is an effort on unit testing so that we can catch when commits break the API.  That will help us reconcile things easier in the future.  I plan on working on tests for the nvd3-community version to catch the breakages in the development branch that keep angularjs-nvd3-directives from working properly.

**Comment by robinfhu** [comment]
@portante yes that will help greatly.  I actually fixed all broken unit tests in my branch, so you can work off there.

**Comment by portante** [comment]
@robinfhu, would you be open to doing that merge work in the development branch of the community?  That way can say that nvd3-community/development and novus/massive-merge branches are up-to-date with each other when we are reviewing?

**Comment by robinfhu** [comment]
@portante I actually merged nvd3-community/development into novus/nvd3/development.  I'm almost done actually, I can PR against nvd3-community.

**Comment by portante** [comment]
@robinfhu, there are differences in the code bases still, so I issued a PR that just took the minimal fixes (staggered axis labels) and test changes so that the community version tests start passing.

I figured that after that, we can take in the rest of the code changes knowing that unit tests are passing.

It seems that we should also add travis-ci integration so that we always run unit tests as part of the production build so that folks can easily see when a change breaks something.

**Comment by liquidpele** [comment]
I commented on portante's PR here:  https://github.com/nvd3-community/nvd3/pull/58 but I'll ask here as well I guess.   

Is that from the changes robin made, or is that different?  I want to avoid conflicts and make sure we're on the same page about what direction this herd of cats is going :)

**Comment by robinfhu** [comment]
Right now my goal is to get novus/nvd3 master synced up as close as possible to nvd3-community. Should just be a few days to fix it up.  Then I have the monster task for going through all the PR's.  

In the meantime, nvd3-community can keep doing its thing. I'll merge in changes to our repo regularly. After a few weeks novus/nvd3 will be in better shape and people will start using it again.

**Comment by robinfhu** [comment]
Also Travis Ci is a great idea.

**Comment by portante** [comment]
@robinfhu, at that point, how will the governance move forward on the novus/nvd3 library?  Will the community be able to approve PRs and help decide what features get added?

**Comment by liquidpele** [comment]
Yes, I'm curious about this too.  I think only one repo should be the main dev one, otherwise they'll just keep diverging due to pull requests going to one or the other but not both.

**Comment by robinfhu** [comment]
Hi guys, so I've been giving this some thought.  The open source community has the advantage of time and enthusiasm to maintain the library, which we want to leverage. Novus' advantage is that we use NVD3 heavily in production an interest in keeping it working.

Basically, yes, we want to add community members to the novus/nvd3 repo.   I'm looking into this right now.

**Comment by joshjordan** [comment]
Hi guys, like @robinfhu said, we think it makes most sense to make
novus/nvd3 the canon repository again, with the ultimate goal of
maintaining only a single repo. The novus repository still gets a lot of
traffic and in an ideal state, I think this is the least confusing for
users. Contributors like you guys are an important part of making that
happen.

Once we get that repo up caught up to the latest community work, I would
really love to talk about the development road map with you guys.

**Comment by robinfhu** [comment]
Okay I updated novus/nvd3 master.  And I added @liquidpele  and @portante as members.  Anyone else who contributed a lot?

**Comment by liquidpele** [comment]
Several people contributed but not sure enough to  Add them, they can always still use pull requests after all..

**Comment by robinfhu** [comment]
@liquidpele yep, can you check to see if you can push to novus/nvd3 now?

**Comment by liquidpele** [comment]
On my phone but i can in an hour.

**Comment by joshjordan** [comment]
Let's stick with you two for now, then, and accept other contributors as we
see necessary in the future. I think we can close this issue now, and
create issues on novus/nvd3 to continue our discussions on how to move
forward.
